File: generators/valkyrieNames.py
__all__ = ['valkyrieNames']

# Don't look below, you will not understand this Python code :) I don't.

from js2py.pyjs import *
# setting scope
var = Scope( JS_BUILTINS )
set_global_object(var)

# Code follows:
var.registers(['names1', 'names2', 'nameGen'])
@Js
def PyJsHoisted_nameGen_(this, arguments, var=var):
    var = Scope({'arguments':arguments, 'this':this}, var)
    var.registers(['result'])
    var.put('result', Js([]))
    #for JS loop
    var.put('i', Js(0.0))
    while (var.get('i')<Js(10.0)):
        try:
            var.put('rnd', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('names1').get('length'))))
            var.put('rnd2', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('names2').get('length'))))
            var.put('names', (var.get('names1').get(var.get('rnd'))+var.get('names2').get(var.get('rnd2'))))
            var.get('result').callprop('push', var.get('names'))
        finally:
                (var.put('i',Js(var.get('i').to_number())+Js(1))-Js(1))
    pass
    return var.get('result')
PyJsHoisted_nameGen_.func_name = 'nameGen'
var.put('nameGen', PyJsHoisted_nameGen_)
var.put('names1', Js([Js('A'), Js('Aga'), Js('Ar'), Js('Bry'), Js('Ey'), Js('Fri'), Js('Fria'), Js('Ge'), Js('Gei'), Js('Go'), Js('Guo'), Js('He'), Js('Hi'), Js('Hja'), Js('Hjo'), Js('Hla'), Js('Hlo'), Js('Hri'), Js('Hru'), Js('Ka'), Js('Mi'), Js('O'), Js('Re'), Js('Regi'), Js('Ro'), Js('Sa'), Js('San'), Js('Si'), Js('Ska'), Js('Ske'), Js('Sko'), Js('Sku'), Js('Sva'), Js('Sve'), Js('Svei'), Js('Svi'), Js('Tho'), Js('Thri'), Js('Thru'), Js('Va')]))
var.put('names2', Js([Js('dana'), Js('dis'), Js('dmadra'), Js('dne'), Js('dr'), Js('dra'), Js('drifa'), Js('dul'), Js('gabi'), Js('gin'), Js('ginleif'), Js('gjold'), Js('grdrifa'), Js('grior'), Js('grun'), Js('gul'), Js('hildr'), Js('hylde'), Js('ja'), Js('la'), Js('ld'), Js('ldana'), Js('ldr'), Js('leif'), Js('lmold'), Js('lna'), Js('lrun'), Js('ma'), Js('madra'), Js('mold'), Js('nd'), Js('ndul'), Js('ngrior'), Js('nhildr'), Js('nhylde'), Js('nul'), Js('pul'), Js('ra'), Js('rdmadra'), Js('rifa'), Js('rior'), Js('rja'), Js('ronul'), Js('run'), Js('rvif'), Js('st'), Js('ta'), Js('tha'), Js('va'), Js('vif')]))
pass
pass


# Add lib to the module scope
valkyrieNames = var.to_python()

File: js/valkyrieNames.js
var names1 = ["A","Aga","Ar","Bry","Ey","Fri","Fria","Ge","Gei","Go","Guo","He","Hi","Hja","Hjo","Hla","Hlo","Hri","Hru","Ka","Mi","O","Re","Regi","Ro","Sa","San","Si","Ska","Ske","Sko","Sku","Sva","Sve","Svei","Svi","Tho","Thri","Thru","Va"];
var names2 = ["dana","dis","dmadra","dne","dr","dra","drifa","dul","gabi","gin","ginleif","gjold","grdrifa","grior","grun","gul","hildr","hylde","ja","la","ld","ldana","ldr","leif","lmold","lna","lrun","ma","madra","mold","nd","ndul","ngrior","nhildr","nhylde","nul","pul","ra","rdmadra","rifa","rior","rja","ronul","run","rvif","st","ta","tha","va","vif"];
var br = "";
function nameGen(){
	var element = document.createElement("div");
	element.setAttribute("id", "result");
	
	for(i = 0; i < 10; i++){
		rnd = Math.floor(Math.random() * names1.length);
		rnd2 = Math.floor(Math.random() * names2.length);
		names = names1[rnd] + names2[rnd2];
		br = document.createElement('br');	
		element.appendChild(document.createTextNode(names));
		element.appendChild(br);
	}
	if(document.getElementById("result")){
		document.getElementById("placeholder").removeChild(document.getElementById("result"));
	}		
	document.getElementById("placeholder").appendChild(element);
}